### PR TITLE
testscript: port "mv" and "! cmp" from upstream

### LINKED ISSUE #153: Port 'cmd/go: add mv and support "! cmp" in script tests'
From https://go-review.googlesource.com/c/go/+/380916

### PR DESCRIPTION
**Pull request by mvdan**
(see commit message)

Fixes #153.

### CODE REVIEW

**Review by rogpeppe — CHANGES_REQUESTED** [review]
LGTM, thanks, except that the new functionality needs to be added to the docs.

**Comment by mvdan** [comment]
PTAL. The way gopls ran 1.19's gofmt, it shifted quite a bit of whitespace in doc.go, which I think we want anyway. To ease your review, ignore whitespace changes: https://github.com/rogpeppe/go-internal/pull/168/files?w=1

File: testscript/doc.go [hunk 1]
```diff
@@ -122,102 +122,107 @@ Additional conditions can be added by passing a function to Params.Condition.
...
+  - mkdir path...
+    Create the listed directories, if they do not already exists.
+
+  - unquote file...
+    Rewrite each file by replacing any leading ">" characters from
+    each line. This enables a file to contain substrings that look like
+    txtar file markers.
+    See also https://godoc.org/github.com/rogpeppe/go-internal/txtar#Unquote
+
+  - mv path1 path2
```
Comment (rogpeppe) [inline testscript/doc.go:189]: Looks like both unquote and mv are out of alphabetical order here (yeah, unquote is an old problem, probably my fault, but worth fixing now too, i reckon).

**Review by rogpeppe — CHANGES_REQUESTED** [review]
LGTM modulo one last trivial thing.

File: testscript/doc.go (hunk 1 above)
Comment (mvdan) [reply testscript/doc.go:183]: hadn't noticed we were trying to keep them sorted :) fixed in https://github.com/rogpeppe/go-internal/compare/24c64e725dba574088acf1dc892bf5fae970160b..2310f38754e627b9db74eba3c5327b73666b4576.

**Review by rogpeppe — APPROVED** [review]
LGTM thanks!

**Pull request merged**